### ISSUE TITLE
no-unused: false positive with substitution type inside mapped type

### ISSUE THREAD

**Issue by ajafff** [ajafff/tslint-consistent-codestyle#85]
```ts
type Values<T> = T extends {[K in keyof T]: infer U} ? U : never;
```

`U` is marked as unused but is actually referenced in the then branch of the conditional type

**Comment by InExtremaRes** [comment]
It's similar but not exactly the same:

```ts
type Foo<T> = T extends { new (a: infer A): any } ? A : never;
```
`A` is marked as unused.

We can make and indirection, though:

```ts
interface Baz<A> { new (a: A): any }
type Bar<T> = T extends Baz<infer A> ? A : never;
```
